src/cert.rs

```diff
@@ -101,10 +101,11 @@ pub(crate) fn parse_cert_internal<'a>(
         // special logic for handling critical Netscape Cert Type extensions.
         // That has been intentionally omitted.
 
-        der::nested(
+        der::nested_ex(
             tbs,
             der::Tag::ContextSpecificConstructed3,
             Error::BadDER,
```

Comment:
In your PR for missing extensions, you'd change this occurrence of `Error::BadDER` to the new error.